uk.co.compucorp.civicrm.hrleaveandabsences/js/test/leave-type-wizard/components/leave-type-wizard.component.spec.js

```diff
@@ -213,6 +238,96 @@ define([
             expect(controller.isOnSectionLastTab).toEqual(true);
           });
         });
+
+        describe('when user cancels the form filling', function () {
+          beforeEach(function () {
+            controller.openSection(0);
+            controller.previousTabHandler();
+            $rootScope.$digest();
+          });
+
+          it('redirects to the leave types list page', function () {
+            expect($window.location.href).toBe(
+              sampleURLOrigin + leaveTypeListPageURL);
+          });
+        });
+
+        describe('when user submits the whole wizard form', function () {
+          describe('basic tests', function () {
+            beforeEach(function () {
+              absenceTypeSaverSpy.and.returnValue($q.resolve());
+              fillWizardIn();
+              submitWizard();
+            });
+
+            it('is loading', function () {
+              expect(controller.loading).toBe(true);
+            });
+
+            describe('when saving finishes', function () {
+              beforeEach(function () {
+                $rootScope.$digest();
+              });
+
+              it('saves absence type', function () {
+                var params = AbsenceType.save.calls.mostRecent().args[0];
```

Comment:
move to `beforeEach` block

Reply:
@reneolivo Not in this case, it was actually @AkA84 who explained me why this should be inside `it` block and not `beforeEach`. The reason is that other `it` blocks don't need this variable and we assign the expression to a variable just for readability sake here. Also it does not make `it` async, which is one of the most important things we discussed earlier. So it is OK and actually good to keep that var inside the `it` block.